### PR TITLE
VideoProperties: fix decoding of first frames

### PR DESCRIPTION
**Pull request by cchampet**

### CODE REVIEW

File: src/AvTranscoder/properties/VideoProperties.cpp [hunk 1]
```diff
@@ -48,7 +48,7 @@ std::string VideoProperties::getProfileName() const
         throw std::runtime_error("unknown codec profile");
```
Comment (valnoel) [inline src/AvTranscoder/properties/VideoProperties.cpp]: You could use _getProfile()_ as well just up here !

File: src/AvTranscoder/properties/FileProperties.cpp [hunk 2]
```diff
@@ -31,10 +31,6 @@ void FileProperties::extractStreamProperties(IProgress& progress, const EAnalyse
 {
     clearStreamProperties();
 
-    // if the analysis level wiil decode some streams parts, seek at the beginning
-    if(level > eAnalyseLevelHeader)
-        const_cast<FormatContext*>(_formatContext)->seek(0, AVSEEK_FLAG_BACKWARD);
-
```
Comment (valnoel) [inline src/AvTranscoder/properties/FileProperties.cpp]: Could you explain why seeking at the beginning of the stream " _make analysis impossible for some AVC Intra media_ " ?

File: src/AvTranscoder/properties/VideoProperties.cpp [hunk 3]
```diff
@@ -20,6 +20,7 @@ namespace avtranscoder
 VideoProperties::VideoProperties(const FormatContext& formatContext, const size_t index, IProgress& progress,
                                  const EAnalyseLevel level)
     : StreamProperties(formatContext, index)
+    , _levelAnalysis(level)
```
Comment (valnoel) [inline src/AvTranscoder/properties/VideoProperties.cpp]: Could be used into the constructor method body, to improve readability:
```c++
if(_levelAnalysis == eAnalyseLevelFirstGop)
    analyseGopStructure(progress);
```

File: src/AvTranscoder/properties/VideoProperties.cpp (hunk 1 above)
Comment (cchampet) [inline src/AvTranscoder/properties/VideoProperties.cpp]: Even better here: 4df7cc4

File: src/AvTranscoder/properties/VideoProperties.cpp (hunk 3 above)
Comment (cchampet) [inline src/AvTranscoder/properties/VideoProperties.cpp]: Done here: 378814d

File: src/AvTranscoder/properties/FileProperties.cpp (hunk 2 above)
Comment (valnoel) [inline src/AvTranscoder/properties/FileProperties.cpp]: My mistake, this is deleted lines.. :)

**Comment by cchampet** [comment]
Too many updates. See PR #247 and #248.

**Pull request closed without merging**